packages/opentelemetry-exporter-collector/src/CollectorExporterBase.ts

```diff
@@ -132,14 +143,106 @@ export abstract class CollectorExporterBase<
     return this._shuttingDownPromise;
   }
 
+  private _getUrl(config: T): string {
+    // if given, use url defined in config
+    if (typeof config.url !== 'undefined') {
+      return config.url;
+    }
+    const endpointFromEnv = getEnv().OTEL_EXPORTER_OTLP_ENDPOINT;
+    const defaultEndpoint = this.getProtocol().includes('http')
+      ? `http://${this._defaultEndpoint}/v1/${this.getExporterType()}s`
+      : this._defaultEndpoint;
+    console.log(this.getExporterType(), this.getProtocol(), defaultEndpoint)
+    switch (this.getExporterType()) {
+      case 'metric': {
+        const url = getEnv().OTEL_EXPORTER_OTLP_METRICS_ENDPOINT;
+        return url && url.length > 0
+          ? url
+          : endpointFromEnv && endpointFromEnv.length > 0
+          ? endpointFromEnv
+          : defaultEndpoint;
+      }
+      case 'trace': {
+        const url = getEnv().OTEL_EXPORTER_OTLP_TRACES_ENDPOINT;
+        return url && url.length > 0
+          ? url
+          : endpointFromEnv && endpointFromEnv.length > 0
+          ? endpointFromEnv
+          : defaultEndpoint;
+      }
+      default: {
+        return getEnv().OTEL_EXPORTER_OTLP_ENDPOINT;
+      }
+    }
+  }
+
+  private _getDefaultServiceName(config: T): string {
+    // if given, use url defined in config
+    if (typeof config.serviceName !== 'undefined') {
+      return config.serviceName;
+    }
+    switch (this.getExporterType()) {
+      case 'metric':
+        return 'collector-metric-exporter';
+      case 'trace':
+        return 'collector-trace-exporter';
+      default:
+        return 'collector-exporter';
+    }
+  }
+
+  private _getHeaders(config: T): Record<string, string> {
```

Comment:
the same with headers, base class is used by all 3 protocols, grpc doesn't need this at all. That's why I think the headers should not be a part of base class